includes/config.php

```diff
@@ -0,0 +1,637 @@
+			'section'     => 'weather',
+			'type'        => 'text'
+		)
+	);
+
+	$wp_customize->add_setting(
+		'weather_service_cache_duration',
+		array (
+			'type' => 'option',
+			'default' => Utilities\get_option_default( 'weather_service_cache_duration' )
+		)
+	);
+	$wp_customize->add_control(
+		'weather_service_cache_duration',
+		array(
+			'label'       => 'Weather Data Cache Duration',
+			'description' => 'How long, in seconds, retrieved weather data should be cached.',
+			'section'     => 'weather',
+			'type'        => 'number'
+		)
+	);
+
+	$wp_customize->add_setting(
+		'weather_service_timeout',
+		array (
+			'type' => 'option',
+			'default' => Utilities\get_option_default( 'weather_service_timeout' )
+		)
+	);
+	$wp_customize->add_control(
+		'weather_service_timeout',
+		array(
+			'label'       => 'Weather Service Timeout',
+			'description' => 'Number of seconds to wait before timing out a weather service request.',
+			'section'     => 'weather',
+			'type'        => 'number'
+		)
+	);
+
+	//
+	// Analytics
+	//
+	$wp_customize->add_setting(
+		'news_utm_source',
+		array (
+			'type' => 'option',
+			'default' => Utilities\get_option_default( 'news_utm_source' )
+		)
+	);
+	$wp_customize->add_control(
+		'news_utm_source',
+		array(
+			'label'       => 'News & Announcements - UTM Source',
+			'description' => 'The UTM "source" value to set on News & Announcement email links.',
+			'section'     => 'analytics',
+			'type'        => 'text'
+		)
+	);
+
+	$wp_customize->add_setting(
+		'news_utm_medium',
+		array (
+			'type' => 'option',
+			'default' => Utilities\get_option_default( 'news_utm_medium' )
+		)
+	);
+	$wp_customize->add_control(
+		'news_utm_medium',
+		array(
+			'label'       => 'News & Announcements - UTM Medium',
+			'description' => 'The UTM "medium" value to set on News & Announcement email links.',
+			'section'     => 'analytics',
+			'type'        => 'text'
+		)
+	);
+
+	$wp_customize->add_setting(
+		'news_utm_campaign',
+		array (
+			'type' => 'option',
+			'default' => Utilities\get_option_default( 'news_utm_campaign' )
+		)
+	);
+	$wp_customize->add_control(
+		'news_utm_campaign',
+		array(
+			'label'       => 'News & Announcements - UTM Campaign',
+			'description' => 'The UTM "campaign" value to set on News & Announcement email links.',
+			'section'     => 'analytics',
+			'type'        => 'text'
+		)
+	);
+}
+
+add_action( 'customize_register', __NAMESPACE__ . '\define_customizer_controls', 10 );
+
+
+/**
+ * Remove old blogroll Links and Comments admin pages.
+ *
+ * @since 3.0.0
+ * @author Jo Dickson
+ * @return void
+ **/
+function kill_unused_admin_pages() {
+	remove_menu_page( 'link-manager.php' );
+	remove_menu_page( 'edit-comments.php' );
+}
+
+add_action( 'admin_menu', __NAMESPACE__ . '\kill_unused_admin_pages' );
+
+
+/**
+ * Convenience template redirects
+ *
+ * @author Chris Conover
+ **/
+function gmucf_template_redirect() {
```

Comment:
It might be nice to have this function in a separate file, so it's a little more obvious where to find it if there's a problem with it.

Reply:
I'm almost inclined to remove these hard-coded template redirects entirely in favor of creating actual email posts for each variant.  But yes, I can move these to a separate file for the time being